### DISCUSSION TITLE
Getting data from a Restful api just like GraphQL at build time?

### DISCUSSION THREAD

**Discussion by Blakeinstein** [gatsbyjs/gatsby#27610] [Help]
I am trying to get a list of repositories and its data using GraphQL. This works flawlessly on Gatsby with its intended graphql usage and the gatsby-source-github-api plugin.

However, the Github graphql api doesnt support getting information about contributors, and one needs to use the Rest Api for the same.

I am unsure how I can get the data at build time instead of when someone visits the webpage.

Currently I have
```js
const projectCards = data.githubData.data.organization.repositories.edges

  return (
    <>
      <div className="body-container">
        <br />
        <br />
        <div className="projects-header">Projects</div>
        <br />
        <Grid container spacing={3}>
          {projectCards.map(edge => {
            let projectCard = edge.node;

            return (
              <Grid item xs={12} sm={6} lg={4}>
                <ProjectCard
                  title={projectCard.name || "Error finding name"}
                  imageURL={projectCard.openGraphImageUrl ||  "https://images.pexels.com/photos/2529973/pexels-photo-2529973.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"}
                  description={projectCard.description || "No description found"}
                  repoLink={projectCard.projectsUrl}
                  contributers={projectCard.contributers} // need contributor info here
                />
              </Grid>
            )
```

since the repository url depends on the list of repositories, I cant exactly use export.createPages in gatsby-node... 

```
requestUrl = `https://api.github.com/repos/orgname/${projectCard.name}/contributors`
```

**Comment by KyleAMathews** [comment]
createPages gets passed a `graphql` function you can use to grab the id and then call the REST api.

**Comment by Blakeinstein** [comment]
But then How would I create cards for the same? There is only one page, that I need to do this for.

Currently I can get most of the data needed and create components with each edge.
